pkg/monitoring/metrics.go

```diff
@@ -40,6 +46,36 @@ func SetReadyGauge(isReady bool) {
 	}
 }
 
+func setKubemacpoolDeployedGauge(isDeployed bool) {
+	if isDeployed {
+		kmpDeployedGauge.Set(1)
+	} else {
+		kmpDeployedGauge.Set(0)
+	}
+}
+
+func TrackMonitoringComponents(conf *cnao.NetworkAddonsConfigSpec, monitoringAvailable bool) error {
+	if !monitoringAvailable {
+		return nil
+	}
+
+	err := trackKubemacpoolMonitoring(conf)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func trackKubemacpoolMonitoring(conf *cnao.NetworkAddonsConfigSpec) error {
+	isKubemacpoolDeployed := false
+	if conf.KubeMacPool != nil {
```

Comment:
conf.KubeMacPool means its actually deployed ?
or just that it appeared on the manifest, and might be broken / just now reconcile started to deploy it ?